mnist/README.md

```diff
@@ -0,0 +1,172 @@
+# Mnist
+
+*This example is a Substra implementation of on [the MNIST example from Keras](https://keras.io/examples/vision/mnist_convnet/), inspired from [Substra's Titanic Example](https://github.com/SubstraFoundation/substra/blob/master/examples/titanic/)*
+
+In this example, we'll see how to setup an objective and how to train algorithms on the MNIST dataset.
+## Prerequisites
+
+In order to run this example, you'll need to:
+
+* use Python 3
+* have [Docker](https://www.docker.com/) installed
+* [install the `substra` cli](https://github.com/SubstraFoundation/substra#install)
+* [install the `substratools` library](https://github.com/substrafoundation/substra-tools)
+* [pull the `substra-tools` docker images](https://github.com/substrafoundation/substra-tools#pull-from-private-docker-registry)
+* create a substra profile to define the substra network to target, for instance:
+    ```sh
+    substra config --profile node-1 http://substra-backend.node-1.com
+    substra login --profile node-1 --username node-1 --password 'p@$swr0d44'
+    ```
+* checkout this repository
+
+All commands in this example are run from the `mnist` folder.
+
+## Data preparation
+
+The first step will be to generate train and test data samples from keras.datasets.mnist
+
+
+To generate the data samples, run:
+```sh
+pip install -r scripts/requirements.txt
+python scripts/generate_data_samples.py
+```
+
+This will create two sub-folders in the `assets` folder:
+* `train_data` contains train data features and labels as numpy array files
+* `test_data` contains test data features and labels as numpy array files
+
+## Writing the objective and data manager
+
+Both objective and data manager will need a proper markdown description, you can check them out in their respective
+folders. Notice that the data manager's description includes a formal description of the data structure.
+
+Notice also that the `metrics.py` and `opener.py` module both rely on classes imported from the `substratools` module.
+These classes provide a simple yet rigid structure that will make algorithms pretty easy to write.
+
+## Writing a simple algorithm
+
+You'll find under `assets/algo_cnn` an implementation of the cnn model in the [Keras example](https://keras.io/examples/vision/mnist_convnet/). Like the metrics and opener scripts, it relies on a
+class imported from `substratools` that greatly simplifies the writing process. You'll notice that it handles not only
+the train and predict tasks but also a lot of data preprocessing.
+
+## Testing our assets
+
+### Using asset command line interfaces
```

Comment:
What is the expected outcome? The reader might wonder

Reply:
Do you mean that i should explain what is each command doing, or that i should copy/paste the output of these commands ?